### ISSUE TITLE
File to import not found or unreadable: breakpoint

### ISSUE THREAD

**Issue by ntreffers** [mhs/scout-app#245]
I confess: I'm a bit new to Scout and Sass, but I'm eager to learn. Unfortunately, an error message appears when I try to set up a Drupal base theme (Zen) using Scout. I'm not sure if Scout is the one to blame, so please forgive me if this issue is not in the right place. As I said, I'm still learning!

These are my settings:

Input folder:
/Users/Niels/Desktop/zentheme/sass

Output folder:
/Users/Niels/Desktop/zentheme/css

This is the error:
`>>> Change detected at 13:18:15 to: styles.scss
error styles.scss (Line 20 of _init.scss: File to import not found or unreadable: breakpoint.
Load paths: /Users/Niels/Desktop/zentheme/sass /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-0.12.2/frameworks/blueprint/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-0.12.2/frameworks/compass/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-960-plugin-0.10.4/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/yui-compass-plugin-0.3/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-h5bp-0.0.5/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/zen-grids-1.2/stylesheets Compass::SpriteImporter) overwrite styles.css 
error chroma-kss-styles.scss (Line 20 of _init.scss: File to import not found or unreadable: breakpoint.
Load paths: /Users/Niels/Desktop/zentheme/sass /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-0.12.2/frameworks/blueprint/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-0.12.2/frameworks/compass/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-960-plugin-0.10.4/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/yui-compass-plugin-0.3/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-h5bp-0.0.5/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/zen-grids-1.2/stylesheets Compass::SpriteImporter) overwrite chroma-kss-styles.css 
error kss-example-chroma.scss (Line 20 of _init.scss: File to import not found or unreadable: breakpoint.
Load paths: /Users/Niels/Desktop/zentheme/sass /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-0.12.2/frameworks/blueprint/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-0.12.2/frameworks/compass/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-960-plugin-0.10.4/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/yui-compass-plugin-0.3/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-h5bp-0.0.5/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/zen-grids-1.2/stylesheets Compass::SpriteImporter) create kss-example-chroma.css 
error kss-only.scss (Line 20 of _init.scss: File to import not found or unreadable: breakpoint.
Load paths: /Users/Niels/Desktop/zentheme/sass /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-0.12.2/frameworks/blueprint/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-0.12.2/frameworks/compass/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-960-plugin-0.10.4/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/yui-compass-plugin-0.3/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/compass-h5bp-0.0.5/stylesheets /Applications/Scout.app/Contents/Resources/vendor/gems/gems/zen-grids-1.2/stylesheets Compass::SpriteImporter) overwrite kss-only.css `

Any help would be appreciated! Can anyone point me in the right direction?

**Comment by TheJaredWilcurt** [comment]
So what's happening here is that Zen requires a Sass Mixin library called "breakpoint" but it doesn't actually come with it. This is pretty common in web development. Rather than bloat your project's repo with a bunch of common dependencies, we just make a list of them in a `package.json` file then use NPM to install the dependencies.

**TLDR:** You are missing a file.

* * *

**Option 1:**

1. Install [Node and NPM](http://nodejs.org)
2. In the command prompt terminal on your machine navigate to the folder with the `package.json` file
3. Run the command `npm install` (this will take a few minutes the first time)
4. Try again with Scout-App 2012

**Option 2:**

1. Download [Scout-App 2016](http://TheJaredWilcurt.github.io/scout-app) (in beta, official version will be released within the next week)
2. It just so happens that Scout-App 2016 comes with the Breakpoint Mixin Library already built in
3. ????
4. Profit

* * *

If you use Scout-App 2016, you should move your project into a project folder called one of these: `github`, `projects`, `repositories`, `repos`, or `websites` and put it in your user profile or My docs folder. Scout-App will auto find them on first launch.

So like one of these:

```
C:\Users\Bob\Projects\Zen
C:\Users\Bob\Documents\GitHub\Zen
~\repos\Zen
~\Documents\Websites\zen
D:\Projects\Zen
C:\Github\Zen
...etc.
```

You get the added benefit of being more organized and having all of your projects in a folder next to each other. Or you can just go to `File > New Project` and point directly to the `zen` folder, that works too.

**Comment by ntreffers** [comment]
Thanks a lot for your answer, your explanation is very clear. I've tried all steps for both options, but got only so far...

When I execute 'npm install', the Zen Starterkit gets extracted to the folder node_modules. So far, so good. But when I run Scout-App 2012 again on this newly extracted folder, I still get the same error concerning Breakpoint. 

`File to import not found or unreadable: breakpoint`

In Scout-App 2016 the error is similar, but not about Breakpoint (probably because it's built in like you said), but with the next Mixin Library that _init.scss asks for. 

`File to import not found or unreadable: chroma/functions`

I guess what this means is that all these Mixin Libraries are simply not installed as they should be when I execute npm install. Maybe there's a problem with the package.json from Zen, but that would be a totally different topic. Or am I missing something here?

Thanks again!

**Comment by TheJaredWilcurt** [comment]
You should be running it from the same location as you did before, but there project wasn't designed to allow for others to pick their own way of processing Sass. It's designed explicitly to have their Gulp file handle it. Gulp is a task runner. If you install it, you can run it from the same folder as your `package.json` by typing either `gulp` or `gulp watch`

To install Gulp:

```
npm install -g gulp
```

This is one of the most common ways of handling Sass for professional web developers. It will completely negate the need for Scout-App or any other Sass GUI. Scout-App exists as a means for those who lack the domain knowledge or time to set up one of these more advanced systems. Since this system is already set up for you, it makes more sense to just use it as designed, rather than trying to modify it to fit another workflow.

**Comment by ntreffers** [comment]
Thank you so much for all the info, you're awesome. I'll set up Gulp and go from there. You've been a great help!
